tests/dialects/test_postgres.py

```diff
@@ -1498,3 +1498,72 @@ def test_locks(self):
         for key_type in ("FOR SHARE", "FOR UPDATE", "FOR NO KEY UPDATE", "FOR KEY SHARE"):
             with self.subTest(f"Test lock type {key_type}"):
                 self.validate_identity(f"SELECT 1 FROM foo AS x {key_type} OF x")
+
+    def test_grant(self):
+        grant_cmds = [
+            "GRANT SELECT ON TABLE users TO role1",
+            "GRANT INSERT, DELETE ON TABLE orders TO user1",
+            "GRANT SELECT ON employees TO manager WITH GRANT OPTION",
+            "GRANT USAGE ON SCHEMA finance TO user2",
+            "GRANT ALL PRIVILEGES ON DATABASE mydb TO PUBLIC",
+            "GRANT CREATE ON SCHEMA public TO developer",
+            "GRANT CONNECT ON DATABASE testdb TO readonly_user",
+            "GRANT TEMPORARY ON DATABASE testdb TO temp_user",
+            "GRANT TRIGGER ON orders TO audit_role",
+            "GRANT REFERENCES ON products TO foreign_key_user",
+            "GRANT TRUNCATE ON logs TO admin_role",
+            "GRANT UPDATE(salary) ON employees TO hr_manager",
+            "GRANT SELECT(id, name), UPDATE(email) ON customers TO customer_service",
+            "GRANT EXECUTE ON FUNCTION calculate_bonus(integer) TO analyst",
+            "GRANT EXECUTE ON FUNCTION CALCULATE_BONUS(integer) TO analyst",
+        ]
+
+        for sql in grant_cmds:
+            with self.subTest(f"Testing PostgreSQL's GRANT statement: {sql}"):
+                self.validate_identity(sql)
+
+        advanced_grants = [
+            "GRANT INSERT, DELETE ON ALL TABLES IN SCHEMA myschema TO user1",
+            "GRANT developer_role TO john",
+            "GRANT admin_role TO mary WITH ADMIN OPTION",
+        ]
+
+        for sql in advanced_grants:
+            with self.subTest(f"Testing PostgreSQL's advanced GRANT statement: {sql}"):
+                self.validate_identity(sql, check_command_warning=True)
```

Comment:
These were not in the original [PR](https://github.com/tobymao/sqlglot/pull/4138), so I added them.